python/pyspark/sql/dataframe.py

```diff
@@ -722,8 +722,6 @@ def __getitem__(self, item):
         [Row(age=5, name=u'Bob')]
         """
         if isinstance(item, basestring):
-            if item not in self.columns:
```

Comment:
or maybe, instead of doing any checks here, is it possible to catch specific errors here thrown by the analyzer and just reformat them nicely?